executor/builder.go

```diff
@@ -1529,33 +1529,108 @@ func (b *executorBuilder) buildIndexLookUpJoin(v *plannercore.PhysicalIndexJoin)
 	if defaultValues == nil {
 		defaultValues = make([]types.Datum, len(innerTypes))
 	}
+	outerKeyCols := make([]int, len(v.OuterJoinKeys))
+	for i := 0; i < len(v.OuterJoinKeys); i++ {
+		outerKeyCols[i] = v.OuterJoinKeys[i].Index
+	}
+	innerKeyCols := make([]int, len(v.InnerJoinKeys))
+	for i := 0; i < len(v.InnerJoinKeys); i++ {
+		innerKeyCols[i] = v.InnerJoinKeys[i].Index
+	}
+	metrics.ExecutorCounter.WithLabelValues("IndexLookUpJoin").Inc()
+
+	// If the inner join keys are the suffix set of the index, then return the IndexLookUpMergeJoin
+	var isIndexScan bool
+	var is *plannercore.PhysicalIndexScan
+	if ir, ok := innerPlan.(*plannercore.PhysicalIndexReader); ok {
```

Comment:
What about `innerPlan.(*plannercore.PhysicalTableReader) && IndexPlans[0].(*PhysicalTableScan)` is true?

Reply:
If code like this? How to get the `IndexScan`?